src/util_test.go

```diff
@@ -9,76 +9,120 @@ import (
 )
 
 func TestConstructMetadata(t *testing.T) {
-	receivedMetadata := map[string]string{
-		"Key1": "Value1",
-		"Key2": "Value2",
-	}
-	contentType := "application/json"
-
-	expectedMetadata := map[string]string{
-		"Content-Type": "application/json",
-		"Key1":         "Value1",
-		"Key2":         "Value2",
-	}
-
-	result := constructMetadata(receivedMetadata, contentType)
-
-	if !reflect.DeepEqual(result, expectedMetadata) {
-		t.Errorf("Unexpected metadata. Expected: %v, but got: %v", expectedMetadata, result)
+	testCases := []struct {
+		name             string
+		receivedMetadata map[string]string
+		contentType      string
+		expectedMetadata map[string]string
+	}{
+		{
+			name: "Metadata created wit hexpected values",
```

Comment:
```suggestion
			name: "Metadata created with expected values",
```

Reply:
Should we make the `contentType` a required field? I think this is pretty important when dealing with the callback to insights. 

What do you think?

Reply:
Fixed the typo

What do you mean by making it required? It is required string param to the `constructMetadata` function

Reply:
Oops, nevermind! Got confused with the `}}` in the code and thought that one of the test cases haven't included the `contentType`.